source/preset.cpp

```diff
@@ -90,24 +96,13 @@ bool SavePreset(std::string_view presetName, OptionCategory category) {
         continue;
       }
 
-      //Create all necessary elements
+      // Create the <setting> element
       XMLElement* newSetting = preset.NewElement("setting");
-      XMLElement* settingName = preset.NewElement("settingName");
-      XMLElement* valueName = preset.NewElement("valueName");
-      XMLText* settingText = preset.NewText(std::string(setting->GetName()).c_str());
-      XMLText* valueText = preset.NewText(setting->GetSelectedOptionText().c_str());
-
-      //Some setting names have punctuation in them. Set all values as CDATA so
-      //there are no conflicts with XML
-      settingText->SetCData(true);
-      valueText->SetCData(true);
-
-      //add elements to the document
-      settingName->InsertEndChild(settingText);
-      valueName->InsertEndChild(valueText);
-      newSetting->InsertEndChild(settingName);
-      newSetting->InsertEndChild(valueName);
-      preset.InsertEndChild(newSetting);
+      newSetting->SetAttribute("name", std::string(setting->GetName()).c_str());
```

Comment:
This looks like a potential dangling pointer, or a needlessly complex statement.

Simplified explanation: the statement `std::string(settings->GetName())` builds a new `std::string` object, that will be destroyed after this line. The `.c_str()` operation returns a memory location that will be freed once that object is destroyed. If `SetAttribute` makes a copy of the value, the call is safe but complex, if it uses a pointer it might dangle.

In both cases, this should fix it:
```c++
newSetting->SetAttribute("name", setting->GetName().c_str());
```
(I'm assuming `settings->GetName()` returns a `std::string` (most likely a const reference), based on line 102. If it returns a character pointer, the `.c_str()` part can be dropped. Using the wrong value will result in a compile error, so it's pretty easy to figure out what you need.

Reply:
`setting->GetName()` returns a `std::string_view`, which seems to not have a `c_str()` method itself, hence the conversion, I assumed. (It was already there, I only compacted the code a little when I changed it to the new XML structure.)

I also looked into the SetAttribute() function of TinyXML, and they seem to always make a copy of the string, so freeing the memory after this line should not be a problem (from what I understood).